unpacker.go

```diff
@@ -245,6 +247,31 @@ EachLayer:
 		"chainID": chainID,
 	}).Debug("image unpacked")
 
+	if rCtx.Deduplicate {
+		// delete references to successfully unpacked layers
+		layersMap := map[string]struct{}{}
+		for _, desc := range layers {
+			layersMap[desc.Digest.String()] = struct{}{}
+		}
+		pinfo, err := cs.Info(ctx, parentDesc.Digest)
```

Comment:
for now, the content digest will be as label on manifest. maybe we can just read the current platform manifest instead of setting parent descriptor here.

Reply:
I agree. For now `parentDesc` should only be the manifest descriptor. But are there easier ways to know the manifest digest rather than passing the descriptor from handler to `unpack` func?

Reply:
Sorry for late reply. I was thinking to update `image.Manifest` or add helper method like that 

https://gist.github.com/fuweid/d62809d91766efea9c343448a94faa7a

That v1 manifest will contains the config and layer's digest ID so that it makes update label easier.

do it make senses to you?

Reply:
Thanks for the comment. But can we do this refactoring in the following-up PR because I'd like to keep the change as minimal as possible?

Reply:
The update of manifest should be handled out of unpacking handler because it is related to platform filter. And also the manifest filter can make the logic clear. 

I think we don't need to introduce the middle state if we all agree to handle it out of unpacking handler.